src/osiris_log.erl

```diff
@@ -1884,6 +1821,61 @@ open_index_read(File) ->
     _ = file:read(Fd, ?IDX_HEADER_SIZE),
     Fd.
 
+scan_idx(IndexFile, Offset, LastChunkInSegment) when is_list(IndexFile) ->
+    IndexFd = open_index_read(IndexFile),
+    Result = scan_idx(IndexFd, Offset, LastChunkInSegment),
+    file:close(IndexFd),
+    Result;
+
+%% no known last chunk -> return the first chunk
+scan_idx(_Fd, _Offset, undefined) ->
+    {0, ?LOG_HEADER_SIZE};
+
+scan_idx(Fd, Offset, LastChunkInSegment) when is_record(LastChunkInSegment, chunk_info) ->
+    case file:read(Fd, ?INDEX_RECORD_SIZE_B) of
+        {ok,
+         <<ChunkId:64/unsigned,
+           _Timestamp:64/signed,
+           _Epoch:64/unsigned,
+           FilePos:32/unsigned>>} ->
+            case Offset < ChunkId of
+                true ->
+                    %% offset is lower than the first chunk in this segment
+                    {ChunkId, FilePos};
+                false ->
+                    case Offset > (LastChunkInSegment#chunk_info.id + LastChunkInSegment#chunk_info.num - 1) of
```

Comment:
I'm not very strict about this but I prefer binding values to using record accessors. Especially when they result in long lines :)